### ISSUE TITLE
CI failure: jumbotest_catalog, Failed to bind networked socket 

### ISSUE THREAD

**Issue by jkosh44** [cmu-db/noisepage#1313]
# Bug Report

## Summary
During CI `jumbotest_catalog` is failing with the message `Failed to bind networked socket: Address already in use` (Full stack trace is copied at the bottom of summary). This is happening in multiple unrelated PRs which makes me think that it's not related to either of the PRs but an existing issues.
http://jenkins.db.cs.cmu.edu:8080/blue/organizations/jenkins/terrier/detail/PR-1243/57/tests
http://jenkins.db.cs.cmu.edu:8080/blue/organizations/jenkins/terrier/detail/PR-1310/7/tests
Though I have not seen this issue on master yet.

This might be related to #1261

Example Stack Trace:
```
Stacktrace
Running jumbotest_catalog, redirecting output into /jenkins/workspace/terrier_PR-1310/build/build/test-logs/jumbotest_catalog.txt (attempt 1/1)
Running main() from gmock_main.cc
[==========] Running 22 tests from 3 test suites.
[----------] Global test environment set-up.
[----------] 2 tests from CatalogSqlTests
[ RUN      ] CatalogSqlTests.OidCheck
[2020-11-09 15:58:29.754] [network_logger] [error] Failed to bind networked socket: Address already in use
unknown file: Failure
C++ exception with description "Failed to bind networked socket." thrown in SetUp().
[  FAILED  ] CatalogSqlTests.OidCheck (3383 ms)
[ RUN      ] CatalogSqlTests.OptimizerHashmapTest
[2020-11-09 15:58:32.037] [network_logger] [info] Listening on networked socket with port 15721 [PID=5264]
[2020-11-09 15:58:32.037] [network_logger] [info] Listening on Unix domain socket with port 15721 [PID=5264]
[2020-11-09 15:58:32.472] [network_logger] [info] Server Closed
[       OK ] CatalogSqlTests.OptimizerHashmapTest (3476 ms)
[----------] 2 tests from CatalogSqlTests (6860 ms total)
[----------] 15 tests from CatalogTests
[ RUN      ] CatalogTests.LanguageTest
[       OK ] CatalogTests.LanguageTest (1641 ms)
[ RUN      ] CatalogTests.ProcTest
[       OK ] CatalogTests.ProcTest (1679 ms)
[ RUN      ] CatalogTests.DatabaseTest
[       OK ] CatalogTests.DatabaseTest (4224 ms)
[ RUN      ] CatalogTests.NamespaceTest
[       OK ] CatalogTests.NamespaceTest (1616 ms)
[ RUN      ] CatalogTests.UserTableTest
[       OK ] CatalogTests.UserTableTest (1623 ms)
[ RUN      ] CatalogTests.UserIndexTest
[       OK ] CatalogTests.UserIndexTest (1260 ms)
[ RUN      ] CatalogTests.CascadingDropTableTest
[       OK ] CatalogTests.CascadingDropTableTest (508 ms)
[ RUN      ] CatalogTests.CascadingDropNamespaceTest
[       OK ] CatalogTests.CascadingDropNamespaceTest (1039 ms)
[ RUN      ] CatalogTests.CascadingDropNamespaceWithIndexOnOtherNamespaceTest
[       OK ] CatalogTests.CascadingDropNamespaceWithIndexOnOtherNamespaceTest (885 ms)
[ RUN      ] CatalogTests.UserSearchPathTest
[       OK ] CatalogTests.UserSearchPathTest (999 ms)
[ RUN      ] CatalogTests.CatalogSearchPathTest
[       OK ] CatalogTests.CatalogSearchPathTest (1867 ms)
[ RUN      ] CatalogTests.NameNormalizationTest
[       OK ] CatalogTests.NameNormalizationTest (1232 ms)
[ RUN      ] CatalogTests.GetIndexesTest
[       OK ] CatalogTests.GetIndexesTest (748 ms)
[ RUN      ] CatalogTests.GetIndexObjectsTest
[       OK ] CatalogTests.GetIndexObjectsTest (816 ms)
[ RUN      ] CatalogTests.DDLLockTest
[       OK ] CatalogTests.DDLLockTest (1311 ms)
[----------] 15 tests from CatalogTests (21461 ms total)
[----------] 5 tests from NameBuilderTests
[ RUN      ] NameBuilderTests.ForeignKeyTest
[       OK ] NameBuilderTests.ForeignKeyTest (23 ms)
[ RUN      ] NameBuilderTests.UniquenKeyTest
[       OK ] NameBuilderTests.UniquenKeyTest (23 ms)
[ RUN      ] NameBuilderTests.NoneTest
[       OK ] NameBuilderTests.NoneTest (11 ms)
[ RUN      ] NameBuilderTests.LongNameTest
[       OK ] NameBuilderTests.LongNameTest (91 ms)
[ RUN      ] NameBuilderTests.TableNameOnlyTest
[       OK ] NameBuilderTests.TableNameOnlyTest (12 ms)
[----------] 5 tests from NameBuilderTests (164 ms total)
[----------] Global test environment tear-down
[==========] 22 tests from 3 test suites ran. (28487 ms total)
[  PASSED  ] 21 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] CatalogSqlTests.OidCheck
 1 FAILED TEST
/jenkins/workspace/terrier_PR-1310/build
Standard Output
Running jumbotest_catalog, redirecting output into /jenkins/workspace/terrier_PR-1310/build/build/test-logs/jumbotest_catalog.txt (attempt 1/1)
Running main() from gmock_main.cc
[==========] Running 22 tests from 3 test suites.
[----------] Global test environment set-up.
[----------] 2 tests from CatalogSqlTests
[ RUN      ] CatalogSqlTests.OidCheck
[2020-11-09 15:58:29.754] [network_logger] [error] Failed to bind networked socket: Address already in use
unknown file: Failure
C++ exception with description "Failed to bind networked socket." thrown in SetUp().
[  FAILED  ] CatalogSqlTests.OidCheck (3383 ms)
[ RUN      ] CatalogSqlTests.OptimizerHashmapTest
[2020-11-09 15:58:32.037] [network_logger] [info] Listening on networked socket with port 15721 [PID=5264]
[2020-11-09 15:58:32.037] [network_logger] [info] Listening on Unix domain socket with port 15721 [PID=5264]
[2020-11-09 15:58:32.472] [network_logger] [info] Server Closed
[       OK ] CatalogSqlTests.OptimizerHashmapTest (3476 ms)
[----------] 2 tests from CatalogSqlTests (6860 ms total)
[----------] 15 tests from CatalogTests
[ RUN      ] CatalogTests.LanguageTest
[       OK ] CatalogTests.LanguageTest (1641 ms)
[ RUN      ] CatalogTests.ProcTest
[       OK ] CatalogTests.ProcTest (1679 ms)
[ RUN      ] CatalogTests.DatabaseTest
[       OK ] CatalogTests.DatabaseTest (4224 ms)
[ RUN      ] CatalogTests.NamespaceTest
[       OK ] CatalogTests.NamespaceTest (1616 ms)
[ RUN      ] CatalogTests.UserTableTest
[       OK ] CatalogTests.UserTableTest (1623 ms)
[ RUN      ] CatalogTests.UserIndexTest
[       OK ] CatalogTests.UserIndexTest (1260 ms)
[ RUN      ] CatalogTests.CascadingDropTableTest
[       OK ] CatalogTests.CascadingDropTableTest (508 ms)
[ RUN      ] CatalogTests.CascadingDropNamespaceTest
[       OK ] CatalogTests.CascadingDropNamespaceTest (1039 ms)
[ RUN      ] CatalogTests.CascadingDropNamespaceWithIndexOnOtherNamespaceTest
[       OK ] CatalogTests.CascadingDropNamespaceWithIndexOnOtherNamespaceTest (885 ms)
[ RUN      ] CatalogTests.UserSearchPathTest
[       OK ] CatalogTests.UserSearchPathTest (999 ms)
[ RUN      ] CatalogTests.CatalogSearchPathTest
[       OK ] CatalogTests.CatalogSearchPathTest (1867 ms)
[ RUN      ] CatalogTests.NameNormalizationTest
[       OK ] CatalogTests.NameNormalizationTest (1232 ms)
[ RUN      ] CatalogTests.GetIndexesTest
[       OK ] CatalogTests.GetIndexesTest (748 ms)
[ RUN      ] CatalogTests.GetIndexObjectsTest
[       OK ] CatalogTests.GetIndexObjectsTest (816 ms)
[ RUN      ] CatalogTests.DDLLockTest
[       OK ] CatalogTests.DDLLockTest (1311 ms)
[----------] 15 tests from CatalogTests (21461 ms total)
[----------] 5 tests from NameBuilderTests
[ RUN      ] NameBuilderTests.ForeignKeyTest
[       OK ] NameBuilderTests.ForeignKeyTest (23 ms)
[ RUN      ] NameBuilderTests.UniquenKeyTest
[       OK ] NameBuilderTests.UniquenKeyTest (23 ms)
[ RUN      ] NameBuilderTests.NoneTest
[       OK ] NameBuilderTests.NoneTest (11 ms)
[ RUN      ] NameBuilderTests.LongNameTest
[       OK ] NameBuilderTests.LongNameTest (91 ms)
[ RUN      ] NameBuilderTests.TableNameOnlyTest
[       OK ] NameBuilderTests.TableNameOnlyTest (12 ms)
[----------] 5 tests from NameBuilderTests (164 ms total)
[----------] Global test environment tear-down
[==========] 22 tests from 3 test suites ran. (28487 ms total)
[  PASSED  ] 21 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] CatalogSqlTests.OidCheck
 1 FAILED TEST
/jenkins/workspace/terrier_PR-1310/build
```

## Environment

**OS**: Ubuntu 20.04 (LTS)
I have not seen this issue happen on a Mac node.

**Compiler**: Clang 8.0, GCC 9.3

## Steps to Reproduce
Run CI builds. This doesn't happen every time so you may need to run them a couple of times. I have not yet been able to reproduce this locally with Ubuntu 20.04 and GCC 9.3.

### Expected Behavior
CI passes

### Actual Behavior
CI fails

**Comment by jkosh44** [comment]
This is now confirmed also happening on master: http://jenkins.db.cs.cmu.edu:8080/blue/organizations/jenkins/terrier/detail/master/793/tests/

**Comment by jkosh44** [comment]
Fixed with #1316
